src/tool/xmeta/cpp/xmeta_idl_reader.h

```diff
@@ -12,95 +12,57 @@
 #include "XlangParserBaseListener.h"
 #include "xlang_model_listener.h"
 #include "models/xmeta_models.h"
+#include "xlang_error.h"
 
 struct ast_to_st_listener;
 
 namespace xlang::xmeta
 {
     struct xmeta_idl_reader : public xlang_model_listener
     {
-        xmeta_idl_reader(std::string_view const& idl_assembly_name)
-        {
-            m_assembly_names.emplace_back(idl_assembly_name);
-            m_current_assembly = m_assembly_names.back();
-        }
-        friend struct ast_to_st_listener;
-
-        size_t read(std::istream& idl_contents, bool disable_error_reporting = false);
-
-        size_t read(std::istream& idl_contents, XlangParserBaseListener& listener, bool disable_error_reporting = false);
+        xmeta_idl_reader(std::string_view const& idl_assembly_name) : m_xlang_model{ idl_assembly_name }
+        {}
 
-        void resolve();
-
-        void reset(std::string_view const& assembly_name);
+        void read(std::istream& idl_contents, bool disable_error_reporting = false);
+        void read(std::istream& idl_contents, XlangParserBaseListener& listener, bool disable_error_reporting = false);
+        void pass1();
```

Comment:
Consider renaming to something like resolve_type_refs

Reply:
Fixing in later PR.